states.js

```diff
@@ -1,289 +1,316 @@
-    constructor(name) {
-        super(name);
-        this.type = 'ConditionEnd';
-    }
+  constructor(name) {
+    super(name);
+    this.type = 'ConditionEnd';
+  }
 }
 class AllergyOnsetState extends BaseState {
-    constructor(name, target_encounter, codes) {
-        super(name);
-        this.type = 'AllergyOnset';
-        this.target_encounter = target_encounter;
-        this.codes = codes;
-    }
-    toJSON() {
-        return {
-            ...super.toJSON(),
-            "target_encounter":this.target_encounter,
-            "codes":this.codes
-        }
-    }
+  constructor(name, targetEncounter, codes) {
+    super(name);
+    this.type = 'AllergyOnset';
+    this.targetEncounter = targetEncounter;
+    this.codes = codes;
+  }
+
+  toJSON() {
+    return {
+      ...super.toJSON(),
+      target_encounter: this.targetEncounter,
+      codes: this.codes,
+    };
+  }
 }
 class AllergyEndState extends BaseState {
-    constructor(name) {
-        super(name);
-        this.type = 'AllergyEnd';
-    }
+  constructor(name) {
+    super(name);
+    this.type = 'AllergyEnd';
+  }
 }
 class MedicationOrderState extends BaseState {
-    constructor(name, codes) {
-        super(name);
-        this.type = 'MedicationOrder';
-        this.codes = codes;
-    }
-    toJSON() {
-        return {
-            ...super.toJSON(),
-            "codes":this.codes
-        }
-    }
+  constructor(name, codes) {
+    super(name);
+    this.type = 'MedicationOrder';
+    this.codes = codes;
+  }
+
+  toJSON() {
+    return {
+      ...super.toJSON(),
+      codes: this.codes,
+    };
+  }
 }
 class MedicationEndState extends BaseState {
-    constructor(name) {
-        super(name);
-        this.type = 'MedicationEnd';
-    }
+  constructor(name) {
+    super(name);
+    this.type = 'MedicationEnd';
+  }
 }
 class CarePlanStartState extends BaseState {
-    constructor(name, codes) {
-        super(name);
-        this.type = 'CarePlanStart';
-        this.codes = codes;
-    }
-    toJSON() {
-        return {
-            ...super.toJSON(),
-            "codes":this.codes
-        }
-    }
+  constructor(name, codes) {
+    super(name);
+    this.type = 'CarePlanStart';
+    this.codes = codes;
+  }
+
+  toJSON() {
+    return {
+      ...super.toJSON(),
+      codes: this.codes,
+    };
+  }
 }
 class CarePlanEndState extends BaseState {
-    constructor(name) {
-        super(name);
-        this.type = 'CarePlanEnd';
-    }
+  constructor(name) {
+    super(name);
+    this.type = 'CarePlanEnd';
+  }
 }
 class ProcedureState extends BaseState {
-    constructor(name, codes, duration) {
-        super(name);
-        this.type = 'Procedure';
-        this.codes = codes;
-        this.duration = duration;
-    }
-    toJSON() {
-        return {
-            ...super.toJSON(),
-            "codes": this.codes,
-            "duration": this.duration
-        }
-    }
+  constructor(name, codes, duration) {
```

Comment:
```suggestion
  constructor(name, codes, duration = undefined) {
```

we can default the duration arg to undefined so it won't appear in the resulting json if not specified